core/src/main/java/org/sonarsource/sonarlint/core/AutoBinding.java

```diff
@@ -0,0 +1,408 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core;
+
+import com.google.common.eventbus.Subscribe;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.sonarsource.sonarlint.core.client.api.util.TextSearchIndex;
+import org.sonarsource.sonarlint.core.clientapi.SonarLintClient;
+import org.sonarsource.sonarlint.core.clientapi.config.binding.AutoBindCandidate;
+import org.sonarsource.sonarlint.core.clientapi.config.binding.BindingConfiguration;
+import org.sonarsource.sonarlint.core.clientapi.config.binding.SuggestAutoBindParams;
+import org.sonarsource.sonarlint.core.clientapi.config.scope.ConfigurationScope;
+import org.sonarsource.sonarlint.core.clientapi.connection.config.AbstractConnectionConfiguration;
+import org.sonarsource.sonarlint.core.clientapi.connection.config.SonarCloudConnectionConfiguration;
+import org.sonarsource.sonarlint.core.clientapi.connection.config.SonarQubeConnectionConfiguration;
+import org.sonarsource.sonarlint.core.clientapi.fs.FindFileByNamesInScopeParams;
+import org.sonarsource.sonarlint.core.clientapi.fs.FindFileByNamesInScopeResponse;
+import org.sonarsource.sonarlint.core.clientapi.fs.FoundFile;
+import org.sonarsource.sonarlint.core.commons.log.SonarLintLogger;
+import org.sonarsource.sonarlint.core.commons.progress.ProgressMonitor;
+import org.sonarsource.sonarlint.core.event.BindingConfigChangedEvent;
+import org.sonarsource.sonarlint.core.event.ConfigurationScopeAddedEvent;
+import org.sonarsource.sonarlint.core.event.ConnectionAddedEvent;
+import org.sonarsource.sonarlint.core.serverapi.ServerApi;
+import org.sonarsource.sonarlint.core.serverapi.component.ServerProject;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+import static org.apache.commons.lang.StringUtils.removeEnd;
+import static org.apache.commons.lang.StringUtils.trimToNull;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+public class AutoBinding {
```

Comment:
The class is a bit big, maybe we could split it ? There are different concerns, e.g. extracting info from the `.properties` files and auto-bind

Reply:
I will do it in another PR